app/src/main/java/org/oppia/android/app/topic/TopicFragmentPresenter.kt

```diff
@@ -61,6 +63,12 @@ class TopicFragmentPresenter @Inject constructor(
       binding.topicToolbarTitle.isSelected = true
     }
 
+    lifecycleSafeTimerFactory.createTimer(25000).observe(fragment.viewLifecycleOwner) {
```

Comment:
What does this `25000` timer exactly do? Can you please explain this how this fixes the issue.

Reply:
Hi Rajat this fix unselects the selected marquee textview exactly 25 seconds later after it's selected, this is on assumption that the marquue loop has ended or is close to ending this prevents repeat looping incase of other scenarios.

Reply:
Hi Ben unselecting it sure does stop the marqueeing, how do we know that it has finished marqueeing? Yes works correctly however if scenario occurs earlier than timer this does not solve the issue.

Reply:
@BenHenning How do you suggest we conditionally unselect it once it's finished.

Reply:
I think I mentioned this in a chat or in email, but we might actually need to look into extending TextView and forming our own basic marquee functionality in order to handle this properly (since my searches also didn't yield any way to actually detect if the marquee finished). This is the main reason why I suspect that this issue might be a bit harder compared to some of the others you've been looking at--it's not actually obvious what the solution is here.

Reply:
Hi @BenHenning myself too, i haven't found any solid docs on ways to detect this. Check out this suggestion to use reflection
https://stackoverflow.com/questions/12392540/fire-event-when-marquee-completes do you think it can work. I have used reflection to try get fields for testing and it didn't work successfully for me.

Reply:
@KevinGitonga reflection is definitely not ideal (besides the performance implications, it's also tricky to get working correctly with Proguard and is broadly considered an antipattern in production applications).

I looked more deeply into the marquee implementation within TextView and am more confident that it shouldn't be too difficult for us to recreate its behavior. I actually don't think we even need to bother with extending TextView.

I think we can introduce our own text marquee logic that just translated the TextView's text over time until the full text has been translated (i.e. the text's width) at which point it is restarted. Android's marquee implementation does a bit more then this (to account for images within the text view, and fading), but we don't need that complexity. It's also important to note that the marquee seems to update no faster than 30 frames per second which is probably for performance reasons, so we should probably replicate that.

Let me know if you have any specific concerns with approach the solution in this way.

Reply:
@BenHenning I have noticed that using translation animation to move text from RTL may be tricky since text will just go over the back navigation not as in Marquee where it disappears before it.

Reply:
While Rajat originally opened this thread, closing it now since we've implemented an entirely different solution that doesn't have the same issue originally identified.